library/runtime_arm64.c

```diff
@@ -0,0 +1,98 @@
+/*
+ *  Runtime detection module for Arm64 CPU feature sets.
+ *
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 ( the "License" ); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+#include "runtime_internal.h"
+
+#if defined(MBEDTLS_RUNTIME_C) && \
+    defined(MBEDTLS_ARCH_IS_ARM64) && \
+    !defined(MBEDTLS_CPU_HAS_FEATURES_ALT)
+
+#include <stdint.h>
+
+#if defined(__linux__)
+#include <sys/auxv.h>
+#else
+#include <arm_acle.h>
+#endif
+
+#if defined(__linux__)
+
+bool mbedtls_cpu_has_features_arm64(unsigned long hwcap, unsigned long hwcap2)
+{
+    if ((getauxval(AT_HWCAP) & hwcap) != hwcap) {
+        return false;
+    }
+    return (getauxval(AT_HWCAP2) & hwcap2) == hwcap2;
+}
+
+#else
+/* Access ID_*_EL1 to get the CPU feature sets.
+ *
+ * See: https://github.com/torvalds/linux/blob/master/Documentation/arm64/elf_hwcaps.rst
```

Comment:
The reference to HWCAP under Linux should not be in the `#else` section, where `__linux__` is specifically not defined

Reply:
The document is not only for linux. It describes the map table from HW_CAPS* to internal register.

Reply:
I removed this paragraph in this PR. 

I am not sure if it should be implemented in library.